src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/ProofCertificateDownload.swift

```diff
@@ -0,0 +1,104 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+public struct ProofCertificateDownload {
+
+    // MARK: - Public
+
+    public func fetchProofCertificate(
+        for healthCertificates: [Base45],
+        with httpService: HTTPServiceProtocol = HTTPService(),
+        completion: @escaping (Result<CBORData?, ProofCertificateFetchingError>
```

Comment:
Why is the data here optional? I would expect either a success with valid `CBORData` or failure for whatever reasons. This was one of the reasons `Result` objects exist to reduce all the clutter of checking for valid outcome.

I suggest to return something like `.failure(.eligibleCertificatesIsEmpty)`.

Reply:
Good point. From the perspective of the tech spec its not an error, if there is no proof certificate.
But fully agree with your point and would propose to add an error case PC_NO_PROOF.
@nickguendling is it ok for you?

Reply:
Not sure if I agree with the suggestion, because in this case I need to duplicate some code from the success case into one special failure case then that is handled like the success case. I think nil is actually a success here and is handled like that in the service. I would prefer to keep the optional for this, although I understand the general point.